mesonbuild/backend/ninjabackend.py

```diff
@@ -2458,12 +2458,18 @@ def generate_rust_compile_rules(self, compiler) -> None:
 
     def generate_swift_compile_rules(self, compiler) -> None:
         rule = self.compiler_to_rule_name(compiler)
-        full_exe = self.environment.get_build_command() + [
-            '--internal',
-            'dirchanger',
-            '$RUNDIR',
-        ]
-        invoc = full_exe + compiler.get_exelist()
+        wd_args = compiler.get_working_directory_args('$RUNDIR')
+
+        if wd_args is not None:
+            invoc = compiler.get_exelist() + ['-working-directory', '$RUNDIR']
```

Comment:
Oops, this should be this of course, otherwise that return value is pointless.

```suggestion
            invoc = compiler.get_exelist() + wd_args
```